satpy/dataset/metadata.py

```diff
@@ -199,7 +200,7 @@ def _all_arrays_equal(arrays):
 
     If the arrays are lazy, just check if they have the same identity.
     """
-    if hasattr(arrays[0], "compute"):
+    if isinstance(arrays[0].data, da.Array):
```

Comment:
Doesn't this break metadata merging for DataArrays? Maybe it would be better to check if they are not regular numpy arrays? Or rather, it is numpy or is a DataArray with numpy underneath?